### ISSUE TITLE
Bug: Modal with `closeTimeoutMS` unrecoverable with Suspense

### ISSUE THREAD

**Issue by h3rmanj** [reactjs/react-modal#982]
### Summary:

Passing `closeTimeoutMS` to an open Modal makes it unrecoverable if it's part of a render that suspends.

### Steps to reproduce:

 1. Create a component that contains an open modal, and something that can suspend
 2. Suspend it after initial render
 3. Modal can't recover

### Expected behavior:

Modal should be open.

### Link to example of issue:

https://codesandbox.io/s/closetimeoutms-and-suspense-nfmelh?file=/src/App.tsx

### Additional notes:

If the `closeTimeoutMS` prop is removed from the codesandbox link, everything works as expected.

**Comment by diasbruno** [comment]
@h3rmanj This is super weird...why is it triggering the fallback if there is still content been displayed?

**Comment by diasbruno** [comment]
It is the lay component that triggers the suspense. Probably, suspense uses conditional rendering on their side and react-modal doesn't play well with it.

**Comment by diasbruno** [comment]
> React-Modal: Cannot register modal instance that's already open in Unknown 

:thinking: it must be failing to remove the instance from the manager.

**Comment by diasbruno** [comment]
My bad...I thought that the `h1` tag in the sandbox would not allow to trigger the suspense, but actually it comes from the LazyComponent.

react-modal has a small manager to deal with nested modals and dangling references to modals.

In this case, since it doesn't trigger the manager logic to remove the instance from the manager,
it will fail when it tries to open a new instance.

You must be getting this log message:

> React-Modal: Cannot register modal instance that's already open in Unknown

**Comment by h3rmanj** [comment]
Right, so if I remove `closeTimeoutMS` ([https://codesandbox.io/s/closetimeoutms-and-suspense-forked-qe46ox?file=/src/App.tsx](codesandbox)), what appears to be expected behavior is actually showing an old instance that should've been removed?

**Comment by diasbruno** [comment]
If suspense is been triggered, than it must create a new instance of the `Modal`, because that one was already "detroyed".

https://github.com/reactjs/react-modal/blob/8a7426898cc53844dd2d3043e16d429a1a4942ad/src/helpers/portalOpenInstances.js#L24-L32
